hyejin/app/src/main/java/com/joocoding/android/app/githubsearch/viewmodel/MainViewModel.kt

```diff
@@ -1,18 +1,58 @@
 package com.joocoding.android.app.githubsearch.viewmodel
 
+import android.util.Log
 import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.joocoding.android.app.githubsearch.model.response.Repository
-import com.joocoding.android.app.githubsearch.repository.GithubRepository
+import androidx.lifecycle.viewModelScope
+import com.joocoding.android.app.githubsearch.model.Event
+import com.joocoding.android.app.githubsearch.model.GithubRepository
+import com.joocoding.android.app.githubsearch.model.Response
+import com.joocoding.android.app.githubsearch.model.data.Repository
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
 
 class MainViewModel : ViewModel() {
 
-    val repositories: LiveData<List<Repository>>
-        get() = GithubRepository.repositories
+    private val githubRepository = GithubRepository()
+
+    private val _repositories = MutableStateFlow<List<Repository>>(emptyList())
+    val repositories: StateFlow<List<Repository>> = _repositories.asStateFlow()
+
+    private val _toastMessage = MutableLiveData<Event<String>>()
+    val toastMessage: LiveData<Event<String>> = _toastMessage
 
     fun getRepository(query: String = "") {
-        GithubRepository.getRepository(query)
+        viewModelScope.launch {
+            runCatching {
+                githubRepository.getRepositories(query)
+
+            }.onSuccess { data ->
+                when (data) {
+                    is Response.Success -> {
+                        _repositories.value = data.data?.repositories ?: emptyList()
+                    }
+                    is Response.Error -> {
+                        data.message?.let { message ->
+                            _toastMessage.postValue(Event(message))
+                        }
+                    }
+
+                }
+
+            }.onFailure { e ->
+                Log.e("getRepository", e.toString())
+            }
```

Comment:
풀어서 변수명 작성해도 될 것 같아요!
```suggestion
            }.onFailure { exception ->
                Log.e("getRepository", exception.toString())
            }
```